### PR TITLE
Fifth commit

### PR DESCRIPTION
**Pull request by VladFedash**

### CODE REVIEW

File: src/test/java/helpers/BaseOperations.java [hunk 1]
```diff
@@ -11,12 +11,12 @@ public BaseOperations(WebDriver driver) {
...
 
     public void clickButton(WebElement element) {
-        waitForElementVisibility(5, element);
+        waitForElementToBeClickableShort(element);
         element.click();
     }
 
     public void clickButton(By by) {
-        waitForElementVisibility(2, driver.findElement(by));
+        waitForElementVisibilityShort(driver.findElement(by));
```
Comment (OleksandrMishchenkoGit) [inline src/test/java/helpers/BaseOperations.java]: you can move WaitTime in enum and add waitTIme parameter to all your wait methods

File: src/test/java/helpers/WaitUtils.java [hunk 2]
```diff
@@ -19,9 +27,52 @@ public void waitForElementVisibility(long timeout, WebElement element) {
...
+    public void waitForElementToBeClickable(long timeout, WebElement element) {
+        WebDriverWait wait = new WebDriverWait(driver, timeout);
+        wait.until(ExpectedConditions.elementToBeClickable(element));
+    }
+
+    public void waitForElementToBeClickableShort(WebElement element) {
+        waitForElementToBeClickable(SHORT_WAITING_TIME, element);
+    }
+
+    public void waitForUrl(long timeout, String str) {
```
Comment (OleksandrMishchenkoGit) [inline src/test/java/helpers/WaitUtils.java]: remove this method. not really a usable thing

File: src/test/java/helpers/WaitUtils.java [hunk 3]
```diff
@@ -19,9 +27,52 @@ public void waitForElementVisibility(long timeout, WebElement element) {
...
+        WebDriverWait wait = new WebDriverWait(driver, timeout);
+        wait.until(ExpectedConditions.urlContains(str));
+    }
+
+    public void waitForUrlLong(String str) {
+        waitForUrl(LONG_WAITING_TIME, str);
+    }
+
+    public void waitForWindowNumber(int numberOfWindows) {
```
Comment (OleksandrMishchenkoGit) [inline src/test/java/helpers/WaitUtils.java]: same

File: src/test/java/tests/SearchResultTests.java [hunk 4]
```diff
@@ -37,16 +43,21 @@ public void checkSearchResultsContainsWordIphone() {
...
-        waitUtils.waitForElementVisibility(5, searchResultsPage.productAmountOnPage);
+        waitUtils.waitForElementVisibilityShort(searchResultsPage.productAmountOnPage);
         int expectedResult = Integer.parseInt(searchResultsPage.productAmountOnPage
                 .getText().replaceAll("[^0-9]", ""));
         assertEquals(searchResultsPage.titleProductList.size(), expectedResult);
+
+        ((JavascriptExecutor) driver).executeScript("window.open(arguments[0])", "https://rozetka.com.ua/");
+        waitUtils.waitForWindowNumber(WINDOWS_COUNT);
+        homePage.inputToSearchFieldAndPressEnter(SEARCH_WORD_PHONE);
+        assertEquals(pageWithLocators.productTitle.getText(), "«" + SEARCH_WORD_PHONE + "»");
```
Comment (OleksandrMishchenkoGit) [inline src/test/java/tests/SearchResultTests.java]: very weird test

**Review by OleksandrMishchenkoGit — CHANGES_REQUESTED** [review]

File: src/test/java/helpers/BaseOperations.java (hunk 1 above)
Comment (VladFedash) [reply src/test/java/helpers/BaseOperations.java]: Changed

**Review by OleksandrMishchenkoGit — APPROVED** [review]

**Pull request merged**